src/pages/index.js

```diff
@@ -73,25 +115,124 @@ addCardButton.addEventListener("click", () => {
   formValidators["card-add-form"].resetValidation();
 });
 
+//Updating profile information
 function handleProfileFormSubmit(inputValues) {
-  userProfileInformation.setUserInfo(inputValues);
-  profileFormPopup.close();
+  profileFormPopup.setLoading(true);
+  api
+    .updateUserInformation(inputValues)
+    .then((result) => {
+      return result.json();
+    })
+    .then((userInformation) => {
+      userProfileInformation.setUserInfo(userInformation);
+      profileFormPopup.setLoading(false);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(profileFormPopup.close());
```

Comment:
In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


**Please, correct it everywhere**